app/src/main/java/com/asfoundation/wallet/ui/onboarding/OnboardingPageChangeListener.kt

```diff
@@ -121,13 +86,73 @@ class OnboardingPageChangeListener internal constructor(private val view: View)
   override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
     lottieView!!.progress =
         position * (1f / ANIMATION_TRANSITIONS) + positionOffset * (1f / ANIMATION_TRANSITIONS)
-    checkBox!!.setOnClickListener {
-      showWarningText(position)
-      showButton(position)
-    }
-    showWarningText(position)
-    showButton(position)
+    checkBox!!.setOnClickListener { handleUI(position) }
     updatePageIndicator(position)
+    handleUI(position)
+  }
+
+  private fun handleUI(position: Int) {
+    if (position < 3) {
```

Comment:
maybe extract if, else if, else actions to methods for better readability
e.g. if(position > 3) showLastPageButtons()
also maybe we could have a field with the number of onboarding pages and the condition would be if(position == lastPageIndex) and if(position < lastPageIndex)